tapdb/burn_tree.go

```diff
@@ -204,221 +157,189 @@ func (bt *BurnUniverseTree) InsertBurns(ctx context.Context,
+
+			// Query leaves matching the burn point and namespace.
+			// ScriptKeyBytes is nil here as we want all script keys
+			// for this burn point.
+			dbLeaves, err := dbtx.QueryUniverseLeaves(
 				ctx, UniverseLeafQuery{
-					Namespace: namespace,
+					MintingPointBytes: burnPointBytes,
+					Namespace:         namespace,
 				},
 			)
-
-			// It's okay if no leaves are found in the namespace.
-			if err != nil && !errors.Is(err, sql.ErrNoRows) {
-				return fmt.Errorf("error querying all "+
-					"leaves for namespace %s: %w",
-					namespace, err)
+			if errors.Is(err, sql.ErrNoRows) {
+				continue
 			}
-			leavesToQuery = dbLeaves
-		}
-
-		// Process the found leaves. We'll generate the inclusion proof
-		// for each leaf, and then construct the final authenticated
-		// resp.
-		for _, dbLeaf := range leavesToQuery {
-			// Decode the stored proof blob.
-			var burnProof proof.Proof
-			err = burnProof.Decode(
-				bytes.NewReader(dbLeaf.GenesisProof),
-			)
 			if err != nil {
-				return fmt.Errorf("unable to decode burn "+
-					"proof: %w", err)
+				return nil, fmt.Errorf("error querying leaves "+
+					"for burn point %v: %w", burnPoint, err)
 			}
 
-			// Reconstruct the LeafKey used for SMT insertion.
-			scriptPub, err := schnorr.ParsePubKey(
-				dbLeaf.ScriptKeyBytes,
-			)
-			if err != nil {
-				return fmt.Errorf("unable to parse script "+
-					"key: %w", err)
-			}
-			scriptKey := asset.NewScriptKey(scriptPub)
+			leavesToQuery = append(leavesToQuery, dbLeaves...)
+		}
+	} else {
+		// If no specific points, query all leaves in the namespace.
+		dbLeaves, err := dbtx.QueryUniverseLeaves(
+			ctx, UniverseLeafQuery{
+				Namespace: namespace,
+			},
+		)
+		if errors.Is(err, sql.ErrNoRows) {
+			// No leaves found is not an error in this case.
+			return nil, sql.ErrNoRows
+		}
+		if err != nil {
+			return nil, fmt.Errorf("error querying all leaves "+
+				"for namespace %s: %w", &uniNamespace, err)
+		}
+		leavesToQuery = dbLeaves
+	}
 
-			leafKey := universe.LeafKey{
-				OutPoint:  burnProof.OutPoint(),
-				ScriptKey: &scriptKey,
-			}
-			smtKey := leafKey.UniverseKey()
+	if len(leavesToQuery) == 0 {
+		// Return sql.ErrNoRows if no leaves were found.
+		return nil, sql.ErrNoRows
+	}
 
-			// Generate the inclusion proof.
-			inclusionProof, err := tree.MerkleProof(ctx, smtKey)
-			if err != nil {
-				return fmt.Errorf("error generating proof "+
-					"for smt key %x: %w", smtKey, err)
-			}
+	return leavesToQuery, nil
+}
 
-			// Construct the final authenticated leaf.
-			authLeaf := &universe.AuthenticatedBurnLeaf{
-				BurnLeaf: &universe.BurnLeaf{
-					UniverseKey: leafKey,
-					BurnProof:   &burnProof,
-				},
-				BurnTreeRoot: root,
-				BurnProof:    inclusionProof,
-			}
-			resultLeaves = append(resultLeaves, authLeaf)
-		}
+// decodeAndBuildAuthBurnLeaf decodes the raw leaf, reconstructs the key, and
+// builds the AuthenticatedBurnLeaf.
+func decodeAndBuildAuthBurnLeaf(dbLeaf UniverseLeaf) (
```

Comment:
So here we're using the:
```
func myFuncName(param, param) (
	returnVal, returnVal) {
```

style, while further down we're using

```
func myFuncName(param, param
) (returnVal, returnVal) {
```

Can we please at least get consistency in terms of this decision? I way prefer the first one, for reasons previously stated.

Reply:
I know it can seem a bit arbitrary, but it's just based on personal aesthetics. I like the second option when we have multiple return values: IMO the delineation between the return value and args is clear. Breaking early at say the second return value is also an option, but I prefer to have them be on the same line mostly.